src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -363,6 +369,14 @@ public void RemoveStream(QuicStream stream)
             }
         }
 
+        public void QueueStreamForDisposal(QuicStream stream)
+        {
+            lock (SyncObj)
+            {
+                _streamDisposals.Add(stream.DisposeAsync().AsTask());
```

Comment:
This is effectively a memory leak for long-lived connections since we're not removing completed tasks

Reply:
It's highly unlikely we'll go with this change as it can lead to problems mentioned here: https://github.com/dotnet/runtime/issues/102571#issuecomment-2126389596
So it's good to be aware of this, but it's not necessary to fix it here at the moment, if we're doing this just to do some benchmarks.